src/nyx/features/brisque_gaussian.h

```diff
@@ -0,0 +1,366 @@
+                result[i * (image_cols_ - 1) + j] = mscn_[i * image_cols_ + j] * mscn_[(i + 1) * image_cols_ + j + 1];
+            }
+        }
+
+        return result;
+    }
+
+    std::vector<double> get_mscn_secondary_diagonal() {
+
+        std::vector<double> result((image_rows_ - 1) * (image_cols_ - 1));
+
+        for (int i = 0; i < image_rows_ - 1; ++i) {
+            for (int j = 0; j < image_cols_ - 1; ++j) {
+                result[i * (image_cols_ - 1) + j] = mscn_[(i + 1) * image_cols_ + j] * mscn_[i * image_cols_ + j + 1];
+            }
+        }
+
+        return result;
+    }
+
+public:
+
+    Mscn(const std::vector<unsigned int>& img, int rows, int cols, 
+         const std::vector<double>& kernel, int k_rows, int k_cols) {
+
+        image_ = img;   
+        image_rows_ = rows;
+        image_cols_ = cols;
+        
+        kernel_ = kernel;
+        kernel_rows_ = k_rows;
+        kernel_cols_ = k_cols;
+
+        calculate_mscn(); // initialize mscn
+
+    }
+
+    std::unordered_map<MscnType, std::vector<double>> get_coefficients() {
+
+        if (coefficients_.size() == 0) {
+            coefficients_[MscnType::mscn] = get_mscn(); 
+            coefficients_[MscnType::horizontal] = get_mscn_horizontal();
+            coefficients_[MscnType::vertical] = get_mscn_vertical();
+            coefficients_[MscnType::main_diagonal] = get_mscn_diagonal();
+            coefficients_[MscnType::secondary_diagonal] = get_mscn_secondary_diagonal();
+        }
+
+        return coefficients_;
+         
+    }
+
+};
+
+enum DistributionSide { left, right };
+
+class AsymmetricGeneralizedGaussian {
+private:
+    std::vector<double> x_;
+    double alpha_;
+
+    std::vector<double> calculate_x(DistributionSide side) {
+
+        std::vector<double> result;
+
+        if (side == DistributionSide::left) {
+
+            for (double val : x_) {
+                if (val < 0) {
+                    result.push_back(val);
+                }
+            }
+
+        } else if (side == DistributionSide::right) {
+
+            for (double val : x_) {
+                if (val >= 0) {
+                    result.push_back(val);
+                }
+            }
+
+        } else {
+            throw std::invalid_argument("Side was not recognized");
+        }
+
+        return result;
+    }
+
+    double sigma(DistributionSide side) {
+
+        std::vector<double> _x_vec;
+
+        if (side == DistributionSide::right) {
+
+            _x_vec = calculate_x(DistributionSide::right);
+
+        } else if (side == DistributionSide::left) {
+
+            _x_vec = calculate_x(DistributionSide::left);
+
+        } else {
+            throw std::invalid_argument("Side was not recognized");
+        }
+
+        double sum = 0.0;
+        for (double val : _x_vec) {
+            sum += val * val;
+        }
+
+        return std::sqrt(sum / _x_vec.size());
+    }
+
+    double mean_squares(std::vector<double> x_vec) {
```

Comment:
need to pass const ref

Reply:
This is a part of the Brisque feature that we are removing for now. I will remove these files now